core/go/internal/sequencer/common/engine_integration.go

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright © 2025 Kaleido, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/LF-Decentralized-Trust-labs/paladin/common/go/pkg/i18n"
+	"github.com/LF-Decentralized-Trust-labs/paladin/common/go/pkg/log"
+	"github.com/LF-Decentralized-Trust-labs/paladin/core/internal/components"
+	"github.com/LF-Decentralized-Trust-labs/paladin/core/internal/msgs"
+	"github.com/LF-Decentralized-Trust-labs/paladin/sdk/go/pkg/pldtypes"
+	"github.com/LF-Decentralized-Trust-labs/paladin/toolkit/pkg/prototk"
+	"github.com/google/uuid"
+	mock "github.com/stretchr/testify/mock"
+)
+
+// This is the subset of the StateDistributer interface from "github.com/LF-Decentralized-Trust-labs/paladin/core/internal/statedistribution"
+// Here we define the subset that we rely on in this package
+
+type Hooks interface {
+	GetBlockHeight() int64
+	GetNodeName() string
+}
+
+type EngineIntegration interface {
+	// WriteLockStatesForTransaction is a method that writes a lock to the states for a transaction
+	WriteLockStatesForTransaction(ctx context.Context, txn *components.PrivateTransaction) error
+
+	ResetTransactions(ctx context.Context, transactionID uuid.UUID)
+
+	GetStateLocks(ctx context.Context) ([]byte, error)
```

Comment:
Feels like this could mean a lot of things - particularly as it returns `[]byte`